src/clear_save_data_screen.c

```diff
@@ -0,0 +1,212 @@
+{
+    {
+        .bg = 0,
+        .charBaseIndex = 0,
+        .mapBaseIndex = 31,
+        .screenSize = 0,
+        .paletteMode = 0,
+        .priority = 0,
+        .baseTile = 0,
+    },
+    {
+        .bg = 3,
+        .charBaseIndex = 0,
+        .mapBaseIndex = 30,
+        .screenSize = 0,
+        .paletteMode = 0,
+        .priority = 1,
+        .baseTile = 0,
+    },
+};
+
+static const struct WindowTemplate sClearSaveTextWindow[] =
+{
+    {
+        .priority = 0,
+        .tilemapLeft = 3,
+        .tilemapTop = 15,
+        .width = 26,
+        .height = 4,
+        .paletteNum = 15,
+        .baseBlock = 11,
+    },
+    DUMMY_WIN_TEMPLATE
+};
+
+static const struct WindowTemplate sClearSaveYesNo[] =
+{
+    {
+        .priority = 0,
+        .tilemapLeft = 3,
+        .tilemapTop = 2,
+        .width = 5,
+        .height = 4,
+        .paletteNum = 15,
+        .baseBlock = 115,
+    }
+};
+
+void CB2_InitClearSaveDataScreen(void)
+{
+    if(SetupClearSaveDataScreen())
+        CreateTask(Task_DoClearSaveDataScreenYesNo, 0);
+}
+
+static void Task_DoClearSaveDataScreenYesNo(u8 taskId)
+{
+    SetWindowBorderStyle(0, 0, 2, 14);
+    PrintTextOnWindow(0, 1, gText_ClearAllSaveData, 0, 1, 0, 0);
+    CreateYesNoMenu(sClearSaveYesNo, 2, 14, 1);
+    gTasks[taskId].func = Task_ClearSaveDataScreenYesNoChoice;
+}
+
+static void Task_ClearSaveDataScreenYesNoChoice(u8 taskId)
+{
+    switch(sub_8198C58())
+    {
+        case 0:
+            FillWindowPixelBuffer(0, 17);
+            PrintTextOnWindow(0, 1, gText_ClearingData, 0, 1, 0, 0);
+            gTasks[taskId].func = Task_ClearSaveData;
+            break;
+        case 1:
+        case -1:
+            PlaySE(SE_SELECT);
+            DestroyTask(taskId);
+            SetMainCallback2(CB2_FadeAndDoReset);
+    }
+}
+
+static void Task_ClearSaveData(u8 taskId)
+{
+    ClearSaveData();
+    DestroyTask(taskId);
+    SetMainCallback2(CB2_FadeAndDoReset);
+}
+
+static void MainCB(void)
+{
+    RunTasks();
+    UpdatePaletteFade();
+}
+
+static void VBlankCB(void)
+{
+    TransferPlttBuffer();
+}
+
+static bool8 SetupClearSaveDataScreen(void)
+{
+    u16 i;
+
+    switch(gMain.state)
+    {
+        case 0:
+        default:
+            SetVBlankCallback(NULL);
+            SetGpuReg(REG_OFFSET_DISPCNT, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_BG0HOFS, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_BG0VOFS, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_BG3HOFS, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_BG3VOFS, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_WIN0H, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_WIN0V, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_WININ, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_WINOUT, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_BLDCNT, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_BLDALPHA, DISPCNT_MODE_0);
+            SetGpuReg(REG_OFFSET_BLDY, DISPCNT_MODE_0);
```

Comment:
Aside from REG_OFFSET_DISPCNT, none of these should be using the DISPCNT_MODE_0 define.